x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/user/HasPrivilegesResponse.java

```diff
@@ -86,10 +134,16 @@ public void readFrom(StreamInput in) throws IOException {
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
         out.writeBoolean(completeMatch);
+        if (out.getVersion().onOrAfter(Version.V_7_0_0)) {
+            out.writeMap(cluster, StreamOutput::writeString, StreamOutput::writeBoolean);
+        }
```

Comment:
I understand this is the oversight you've mentioned

Reply:
Yes - I actually noticed it while working on application privileges, but forgot to come back and fix it.